### DISCUSSION TITLE
Analítica de datos en Odoo

### DISCUSSION THREAD

**Discussion by joseluisbossio** [OdooAR/odoo-argentina-doc#2] [Ideas]
Hoy sabemos que es necesario tener separado los datos transaccionales de los datos analíticos, por un tema de que cada dia son mas complejas las consultas a la base de datos, computacionalmente hablando.

La idea consiste en contar con extractores de datos a modo de tener un Odoo para analítica de datos, separado del Odoo transaccional (ERP) de modo que las consultas del tipo Inteligencia Empresarial (Business Intelligence) se ejecuten en una instancia separada.

Esto nos va a permitir no sobrecargar al Odoo transaccional con consultas costosas a la base de datos como las que se requieren en analítica de datos.

Otro punto muy importante, para que funcione y sea útil se requiere que sea simple el armado de dashboard, así como un constructor gráfico de reportes.

**Comment by ibuioli** [comment]
Bien, para abrir la discusión al respecto, la idea es conectar un Odoo con otro, y que ese se encargue del procesamiento de datos para analítica. Habría que plantear primero el alcance tecnológico de la conexión antes que el gráfico (que serán tecnologías de Odoo). Para eso seguramente nos limitemos a tres aspectos: XMLRPC, GraphQL o API.

También abro el debate de si conviene que sea otro Odoo o una app realizada en, por ejemplo, NodeJS; que genere su base de datos basándose en la base de datos de Odoo ERP. Y ya con eso genere los reportes y la dashboard. 
Ahí abrimos otra discusión: ¿qué lenguaje o tecnología sería ideal para procesamiento de datos a nivel analítica? ¿Python vs Node vs algo-más?

**Comment by joseluisbossio** [comment]
Para analítica de datos, yo me quedaría en Python. [Data Analysis with Python - Full Course for Beginners](https://youtu.be/r-uOLxNrNk8)
